_layouts/default.html

```diff
@@ -21,7 +21,14 @@
     <meta name="description" content="Sample illustrating the use of {{ page.feature_name }}.">
     <meta name="viewport" content="width=device-width, initial-scale=1">
     <title>{% if page.title %}{{ page.title }}{% else %}{{ page.feature_name }} Sample{% endif %}</title>
-    {% if page.https_redirect %}{% include https_redirect.html %}{% endif %}
+    <script>
+      // If we're running on a real web server (as opposed to localhost on a custom port,
+      // which is whitelisted), then change the protocol to HTTPS.
+      // See https://goo.gl/lq4gCo for an explanation as to why this is needed for some features.
+      if ((!location.port || location.port == "80") && location.protocol != 'https:') {
```

Comment:
I'd argue it's best practise to perform a whitelist than a blacklist. 
In this case, we want to redirect to `https` only if protocol is `http`.
If it's `file:`, `ftp` or whatever "protocol", it shouldn't do a redirect.

I'm nitpicking there.
Feel free to ignore this and merge it ;)